packages/oft-hyperliquid-evm/contracts/library/HyperLiquidComposerCodec.sol

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+import { IHyperLiquidComposer } from "../interfaces/IHyperLiquidComposer.sol";
+import { IHyperLiquidComposerErrors, ErrorMessage } from "../interfaces/IHyperLiquidComposerErrors.sol";
+import { IHyperAsset, IHyperAssetAmount } from "../interfaces/IHyperLiquidComposerCore.sol";
+import { OFTComposeMsgCodec } from "@layerzerolabs/oft-evm/contracts/libs/OFTComposeMsgCodec.sol";
+library HyperLiquidComposerCodec {
+    /// @dev The base asset bridge address is the address of the HyperLiquid L1 contract
+    /// @dev This is the address that the OFT contract will transfer the tokens to when we want to send tokens to HyperLiquid L1
+    /// @dev https://hyperliquid.gitbook.io/hyperliquid-docs/for-developers/hyperevm/hypercore-less-than-greater-than-hyperevm-transfers#system-addresses
+    address public constant BASE_ASSET_BRIDGE_ADDRESS = 0x2000000000000000000000000000000000000000;
+    uint256 public constant BASE_ASSET_BRIDGE_ADDRESS_UINT256 = uint256(uint160(BASE_ASSET_BRIDGE_ADDRESS));
+
+    event OverflowDetected(uint64 amountCore, uint64 maxTransferableAmount);
+
+    /// @notice Converts a core index id to an asset bridge address
+    /// @notice This function is called by the HyperLiquidComposer contract
+    ///
+    /// @param _coreIndexId The core index id to convert
+    ///
+    /// @return _assetBridgeAddress The asset bridge address
+    function into_assetBridgeAddress(uint256 _coreIndexId) internal pure returns (address) {
```

Comment:
NIT:  why use underscores?  It technically isn't "wrong" but may improve readability to use naming such as `convertToAssetBridgeAddress`.